BDArmory/Control/MissileFire.cs

```diff
@@ -8658,105 +8690,103 @@ public void PointDefenseTurretFiring()
             MslTurrets.Clear();
             //missileTarget = null;
             bool firedMissile = false;
-            int APScount = 0;
+            int APScount = pointDefenseWeaponArray.Length;
             int missileCount = 0;
             TargetInfo interceptiontarget = null;
             Vector3 closestTarget = Vector3.zero;
             Vector3 kbCorrection = BDKrakensbane.IsActive ? BDKrakensbane.FloatingOriginOffsetNonKrakensbane : Vector3.zero; // Correction for Krakensbane for bullets and rockets, which haven't been updated yet.
-            using (var weapon = VesselModuleRegistry.GetModules<ModuleWeapon>(vessel).GetEnumerator())
-                while (weapon.MoveNext())
-                {
-                    if (weapon.Current == null) continue;
-                    if (weapon.Current.isAPS || weapon.Current.dualModeAPS)
+            foreach (var weapon in pointDefenseWeaponArray)
+            {
+                if (weapon == null) continue;
+                //if (weapon.isAPS || weapon.dualModeAPS)
```

Comment:
This previously included dual-mode APS, but `pointDefenseWeaponArray` excludes dual-mode APS.
Also below on line 8810.